benchmarktests/target_auth_gcp.go

```diff
@@ -0,0 +1,294 @@
+	BoundServiceAccounts []string `hcl:"bound_service_accounts,optional"`
+	BoundProjects        []string `hcl:"bound_projects,optional"`
+	BoundZones           []string `hcl:"bound_zones,optional"`
+	BoundRegions         []string `hcl:"bound_regions,optional"`
+	BoundInstanceGroups  []string `hcl:"bound_instance_groups,optional"`
+	BoundLabels          []string `hcl:"bound_labels,optional"`
+	MaxJWTExp            string   `hcl:"max_jwt_exp,optional"`
+	TokenBoundCIDRs      []string `hcl:"token_bound_cidrs,optional"`
+	TokenExplicitMaxTTL  string   `hcl:"token_explicit_max_ttl,optional"`
+	TokenMaxTTL          string   `hcl:"token_max_ttl,optional"`
+	TokenNoDefaultPolicy bool     `hcl:"token_no_default_policy,optional"`
+	TokenNumUses         int      `hcl:"token_num_uses,optional"`
+	TokenPolicies        []string `hcl:"token_policies,optional"`
+	TokenPeriod          string   `hcl:"token_period,optional"`
+	TokenTTL             string   `hcl:"token_ttl,optional"`
+	TokenType            string   `hcl:"token_type,optional"`
+}
+
+func (g *GCPAuth) ParseConfig(body hcl.Body) error {
+	g.config = &GCPTestConfig{
+		Config: &GCPAuthTestConfig{
+			GCPAuthConfig:     &GCPAuthConfig{},
+			GCPTestRoleConfig: &GCPTestRoleConfig{},
+		},
+	}
+
+	diags := gohcl.DecodeBody(body, nil, g.config)
+	if diags.HasErrors() {
+		return fmt.Errorf("error decoding to struct: %v", diags)
+	}
+
+	return nil
+}
+
+func (g *GCPAuth) Target(client *api.Client) vegeta.Target {
+	return vegeta.Target{
+		Method: GCPAuthTestMethod,
+		URL:    client.Address() + g.pathPrefix + "/login",
+		Header: g.header,
+		Body:   []byte(fmt.Sprintf(`{"role": "%s", "jwt": "%s"}`, g.roleName, g.jwt)),
+	}
+}
+
+func (g *GCPAuth) Cleanup(client *api.Client) error {
+	g.logger.Trace(cleanupLogMessage(g.pathPrefix))
+	_, err := client.Logical().Delete(strings.Replace(g.pathPrefix, "/v1/", "/sys/", 1))
+	if err != nil {
+		return fmt.Errorf("error cleaning up mount: %v", err)
+	}
+	return nil
+}
+
+func (g *GCPAuth) GetTargetInfo() TargetInfo {
+	return TargetInfo{
+		method:     GCPAuthTestMethod,
+		pathPrefix: g.pathPrefix,
+	}
+}
+
+func (g *GCPAuth) Setup(client *api.Client, randomMountName bool, mountName string) (BenchmarkBuilder, error) {
+	var err error
+	authPath := mountName
+	config := g.config.Config
+	g.logger = targetLogger.Named(GCPAuthTestType)
+
+	if randomMountName {
+		authPath, err = uuid.GenerateUUID()
+		if err != nil {
+			return nil, fmt.Errorf("can't generate UUID for mount name: %v", err)
+		}
+	}
+
+	g.logger.Trace(mountLogMessage("auth", "gcp", authPath))
+	err = client.Sys().EnableAuthWithOptions(authPath, &api.EnableAuthOptions{
+		Type: "gcp",
+	})
+	if err != nil {
+		return nil, fmt.Errorf("error enabling gcp: %v", err)
+	}
+	setupLogger := g.logger.Named(authPath)
+
+	// check if the provided argument should be read from file
+	creds := config.GCPAuthConfig.Credentials
+	if len(creds) > 0 && creds[0] == '@' {
+		var err error
```

Comment:
Doesn't look like its needed since the next line uses `:=`?
```suggestion
		var err error
```